delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutesSpec.scala

```diff
@@ -163,10 +194,29 @@ class ResourcesRoutesSpec extends BaseRouteSpec with IOFromMap {
           response.asJson shouldEqual
             resourceMetadata(projectRef, id, schema, (nxv + "Custom").toString, createdBy = alice, updatedBy = alice)
         }
+      }
+    }
 
+    "create a tagged resource with an authenticated user and provided id" in {
+      val endpoints           = List(
+        ("/v1/resources/myorg/myproject/_/myid8?tag=mytag", myId8, schemas.resources),
+        ("/v1/resources/myorg/myproject/myschema/myid9?tag=mytag", myId9, schema1.id)
+      )
+      val (routes, resources) = routesWithDecodingOption(DecodingOption.Strict)
+      forAll(endpoints) { case (endpoint, id, schema) =>
+        val payload = jsonContentOf("resources/resource.json", "id" -> id)
+        Put(endpoint, payload.toEntity) ~> asAlice ~> routes ~> check {
+          status shouldEqual StatusCodes.Created
+          response.asJson shouldEqual
+            resourceMetadata(projectRef, id, schema, (nxv + "Custom").toString, createdBy = alice, updatedBy = alice)
+          resourceTags(resources, id).contains(tag) shouldBe true
+        }
       }
     }
 
+    def resourceTags(resources: Resources, id: Iri): Tags =
+      resources.fetch(ResourceRef(id), projectRef).accepted.value.tags
```

Comment:
I think even better than this would be to look up by tag. I know this is similar, but lookup is the reason we have this functionality